src/server/services/pipeline-in-progress/processData.js

```diff
@@ -69,16 +73,57 @@ async function processData(rows, meterID, timeSort = TimeSortTypesJS.increasing,
 	// Convert readingGap and readingLengthVariation to milliseconds to stay consistent with moment.diff() which returns the difference in milliseconds
 	const msReadingGap = readingGap * 1000;
 	const msReadingLengthVariation = readingLengthVariation * 1000;
-	// Retrieve and set the last reading stored for the meter
-	// TODO: Create a redux state to hold these values with other meter states
+	// The following variables are only used if honorDst is true.
+	// Holds the endTimeStamp of the previous reading when we cross out of DST.
+	let prevEndTimestamp;
+	// Holds the original endTimestamp of readings since it is sometimes shifted.
+	let origEndTimestamp;
+	// Tells if processing DST data across readings.
+	let inDst = false;
+	// Tells if just stopped inDst with this reading.
+	let inDstStop = false;
+	// The start time to use for saved reading if first reading to accept after backward shift out of daylight savings time.
+	let startTimestampUse;
+	// Going to need to know meter timezone if honoring DST
+	let meterZone;
+	// The shift in time when a DST crossing occurs.
+	let shift;
+	// Tells if we need to split this reading.
+	let splitDst = false;
+	// Date/time daylight savings starts/ends when a crossing occurs.
+	// The array holding this in moment-timezone is called untils.
+	let zoneUntil;
+	// The start and end timestamp in the timezone of the provided meter.
+	// Retrieve and set the last reading and info stored for the meter
+	// TODO We have an issue that this will not return all the values unless admin but we have a CSV login.
 	const meter = await Meter.getByID(meterID, conn);
 	let meterReading = meter.reading;
 	// These reading values are needed for cumulative data.
 	let meterReading1 = meter.reading;
 	let meterReading2 = meter.reading;
-	let startTimestamp = meter.startTimestamp;
-	let endTimestamp = meter.endTimestamp;
+	// These are the start and end times in the timezone they were saved and in UTC.
+	// Note the meter stores as a string to preserve the shift.
+	// The use of timezone here is a misnomer because this is really the offset.
+	// See below on why create timestamps this way.
+	let startTimestampTz = moment.parseZone(meter.startTimestamp, true);
+	let endTimestampTz = moment.parseZone(meter.endTimestamp, true);
+	startTimestamp = moment.parseZone(startTimestampTz.clone()).tz('UTC', true);
+	endTimestamp = moment.parseZone(endTimestampTz.clone()).tz('UTC', true);
```

Comment:
Thanks. I have added the lets. I think I originally had them declared about during earlier work and then removed it so the let was lost. Good catch.
I debated what to do about the endTimestampMeterZone since it is used in three places. I decided to put a const in front of each use inside each if rather than a let at a higher scope.